exporter/clickhousetracesexporter/clickhouse_exporter_v3.go

```diff
@@ -0,0 +1,360 @@
+			} else {
+				hash = md5.Sum([]byte(span.ServiceName + errorEvent.Event.AttributeMap["exception.type"] + errorEvent.Event.AttributeMap["exception.message"]))
+			}
+			errorEvent.ErrorGroupID = fmt.Sprintf("%x", hash)
+		}
+		stringEvent, _ := json.Marshal(event)
+		span.Events = append(span.Events, string(stringEvent))
+		span.ErrorEvents = append(span.ErrorEvents, errorEvent)
+	}
+}
+
+type attributesData struct {
+	StringMap      map[string]string
+	NumberMap      map[string]float64
+	BoolMap        map[string]bool
+	SpanAttributes []SpanAttribute
+}
+
+func (attrMap *attributesData) add(key string, value pcommon.Value) {
+	spanAttribute := SpanAttribute{
+		Key:      key,
+		TagType:  "tag",
+		IsColumn: false,
+	}
+
+	if value.Type() == pcommon.ValueTypeDouble {
+		attrMap.NumberMap[key] = value.Double()
+		spanAttribute.NumberValue = value.Double()
+		spanAttribute.DataType = "float64"
+	} else if value.Type() == pcommon.ValueTypeInt {
+		attrMap.NumberMap[key] = float64(value.Int())
+		spanAttribute.NumberValue = float64(value.Int())
+		spanAttribute.DataType = "float64"
+	} else if value.Type() == pcommon.ValueTypeBool {
+		attrMap.BoolMap[key] = value.Bool()
+		spanAttribute.DataType = "bool"
+	} else if value.Type() == pcommon.ValueTypeMap {
+		// flatten map
+		result := flatten.FlattenJSON(value.Map().AsRaw(), "")
+		for tempKey, tempVal := range result {
+			tSpanAttribute := SpanAttribute{
+				Key:      tempKey,
+				TagType:  "tag",
+				IsColumn: false,
+			}
+			switch tempVal := tempVal.(type) {
+			case string:
+				attrMap.StringMap[tempKey] = tempVal
+				tSpanAttribute.StringValue = tempVal
+				tSpanAttribute.DataType = "string"
+			case float64:
+				attrMap.NumberMap[tempKey] = tempVal
+				tSpanAttribute.NumberValue = tempVal
+				tSpanAttribute.DataType = "float64"
+			case bool:
+				attrMap.BoolMap[tempKey] = tempVal
+				tSpanAttribute.DataType = "bool"
+			}
+			attrMap.SpanAttributes = append(attrMap.SpanAttributes, tSpanAttribute)
+		}
+		return
+	} else {
+		attrMap.StringMap[key] = value.AsString()
+		spanAttribute.StringValue = value.AsString()
+		spanAttribute.DataType = "string"
+	}
+	attrMap.SpanAttributes = append(attrMap.SpanAttributes, spanAttribute)
+}
+
+func newStructuredSpanV3(bucketStart uint64, fingerprint string, otelSpan ptrace.Span, ServiceName string, resource pcommon.Resource, config storageConfig) (*SpanV3, error) {
+	durationNano := uint64(otelSpan.EndTimestamp() - otelSpan.StartTimestamp())
+
+	isRemote := "unknown"
+	flags := otelSpan.Flags()
+	if flags&hasIsRemoteMask != 0 {
+		isRemote = "no"
+		if flags&isRemoteMask != 0 {
+			isRemote = "yes"
+		}
+	}
+
+	attrMap := attributesData{
+		StringMap:      make(map[string]string),
+		NumberMap:      make(map[string]float64),
+		BoolMap:        make(map[string]bool),
+		SpanAttributes: []SpanAttribute{},
+	}
+
+	resourceAttrs := map[string]string{}
+
+	otelSpan.Attributes().Range(func(k string, v pcommon.Value) bool {
```

Comment:
I do not know how many have nested resource attributes, but it would be nice to have flattened maps everywhere.